pal_statistics/CMakeLists.txt

```
@@ -4,9 +4,22 @@ project(pal_statistics)
 find_package(ament_cmake_auto REQUIRED)
 ament_auto_find_build_dependencies()
 
+if(WIN32)
+  # set the same behavior for windows as it is on linux
+  set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+
+  find_package(Boost QUIET COMPONENTS headers)
+  if(NOT Boost_headers_FOUND)
+    find_package(Boost REQUIRED)
+  endif()
+endif()
+
 ament_auto_add_library(${PROJECT_NAME} SHARED
         src/pal_statistics_utils.cpp src/pal_statistics.cpp
         src/pal_statistics_macros.cpp src/registration_list.cpp)
+target_link_libraries(${PROJECT_NAME}
+        Boost::boost
+)
```

```suggestion
if(WIN32)
  target_link_libraries(${PROJECT_NAME}
          Boost::boost
  )
endif()
```
Otherwise you're breaking Linux build

---

@christophfroehlich the `ament_auto_find_build_dependencies`  somehow works in Linux.

Do we really need to link with `Boost::boost`?

---

https://github.com/ros-controls/ros2_control/actions/runs/15997105245/job/45226155943
I can confirm that you suggestion works on windows+rolling.
Without the if it worked on noble-rolling, but had build failures with jammy-humble like you had.